lib/task.js

```diff
@@ -599,10 +598,11 @@ export default class Task {
       return await this.abortRun(this.taskState);
     }
     // Validate the schema!
-    let payloadErrors =
-      this.runtime.schema.validate(this.task.payload, PAYLOAD_SCHEMA);
+    let payloadErrors = await validator().then(function (validator) {
+      return validator.check(this.task.payload, payloadSchema);
```

Comment:
Try ```js
() => {return validator.check(this.task.payload, payloadSchema);
}```
in place of the anonymous function. As it stands, (I believe) the contextual `this` may be set to the wrong thing.